crates/biome_lsp/src/session.rs

```diff
@@ -492,6 +492,20 @@ impl Session {
         if let Some(config_path) = &self.config_path {
             let base_path = ConfigurationPathHint::FromUser(config_path.clone());
             let status = self.load_biome_configuration_file(base_path).await;
+            self.set_configuration_status(status);
+        } else if let Some(config_path) = self
```

Comment:
We need to consider how to migrate users (LSP clients) smoothly for this change. For example, biome-intellij currently using `--config-path` for overriding the config path. To prefer this new option on the plugin, we need to check the Biome version is >= 2.0. This requires some additional work for plugin side and not a smooth path.

Maybe should we prefer this new option _over_ the existing `--config-path` flag? If so, we can just put the config path to both the CLI flag and the LSP request to support Biome < 2 and >= 2.

Reply:
Since we are about to do a v2 release, I believe this is the right time to remove stuff. I know that users would prefer to have a deprecation warning, but I wouldn't want to carry `--config-path` until we hit v3.

I believe this is the right time to have core and editor extensions to do the breaking change. VSCode extension has already removed it. We should do the same for IntelliJ and Zed.

Reply:
Agree. I will rebase this branch and remove `--config-path` option. Later, I will also remove from biome-intellij, and find a chance for Zed.

Reply:
I will remove `--config-path` in a separated PR.